modules/llrt_stream/src/readable.rs

```diff
@@ -280,17 +289,27 @@ where
                                             break;
                                         }
 
-                                        Self::emit_str(
-                                            This(this2.clone()),
-                                            &ctx3,
-                                            "data",
-                                            vec![Buffer(buffer.clone()).into_js(&ctx3)?],
-                                            false
-                                        )?;
+                                        if cb.is_some() {
+                                            if let Some(buf) = &combined_std_buffer {
+                                                let mut stdout_lock = buf.lock().unwrap();
+                                                stdout_lock.extend_from_slice(&buffer);
+                                            }
+                                            this2.borrow_mut().inner_mut().state = ReadableState::Flowing;
+                                        }
+                                        if let Some(listener) = listener {
+                                            if listener == "data" {
+                                                Self::emit_str(
+                                                    This(this2.clone()),
+                                                    &ctx3,
+                                                    "data",
+                                                    vec![Buffer(buffer.clone()).into_js(&ctx3)?],
+                                                    false
+                                                )?;
+                                            }
+                                        }
```

Comment:
This is a change in behaviour for which I not qualified to comment, @richarddavison